### PR TITLE
TSK-1857: Move taskana-core tests into a dedicated module

### PR DESCRIPTION
**Pull request by ryzheboka**
Changed documentation: https://taskana.atlassian.net/wiki/spaces/TAS/pages/1821935692/Continuous+Integration+GitHub+Actions and https://taskana.atlassian.net/wiki/spaces/TAS/pages/1758625810/Module+Overview

Sonarcloud: https://sonarcloud.io/summary/new_code?id=ryzheboka_taskana&branch=TSK-1857

---
 
### For the submitter:
- [x] I updated the [documentation](https://taskana.atlassian.net/wiki/spaces/TAS/overview) and will supply links to the specific files
- [ ] I did not update the [documentation](https://taskana.atlassian.net/wiki/spaces/TAS/overview)
- [x] I included a link to the [SonarCloud branch analysis](https://taskana.atlassian.net/wiki/spaces/TAS/pages/1019969636/SonarCloud+Integration)
- [ ] I added a description of changes on the [current release notes](https://taskana.atlassian.net/wiki/spaces/TAS/pages/1281392672/Current+Release+Notes+Taskana)
- [x] I did not update the [current release notes](https://taskana.atlassian.net/wiki/spaces/TAS/pages/1281392672/Current+Release+Notes+Taskana)
- [x] I put the ticket in review
- [ ] After integration of the pull request, I verified our [bluemix test environment](http://taskana.mybluemix.net/taskana) is not broken

### Verified by the reviewer:
- [x] Commit message format → TSK-XXX: Your commit message.
- [x] Submitter's update to [documentation](https://taskana.atlassian.net/wiki/spaces/TAS/overview) is sufficient
- [ ] SonarCloud analysis meets our standards
- [ ] Update of the [current release notes](https://taskana.atlassian.net/wiki/spaces/TAS/pages/1281392672/Current+Release+Notes+Taskana) reflects changes
- [ ] PR fulfills the ticket
- [ ] Edge cases and unwanted side effects are tested
- [ ] Readability

### CODE REVIEW

File: lib/taskana-core-test/src/test/java/acceptance/task/query/TaskQueryImplAccTest.java [hunk 1]
```diff
@@ -262,6 +260,7 @@ void should_FilterByExternalId_When_QueryingForExternalIdNotIn() {
     @Nested
     @TestInstance(Lifecycle.PER_CLASS)
     class Received {
+
```
Comment (mustaphazorgati) [inline lib/taskana-core-test/src/test/java/acceptance/task/query/TaskQueryImplAccTest.java:263]: Did you add this manually, or did you IDE format this on its own?

Comment (mustaphazorgati) [inline lib/taskana-core-test/src/test/java/acceptance/task/query/TaskQueryImplAccTest.java:263]: Did you add this manually, or did you IDE format this on its own?

File: lib/taskana-core-test/src/test/resources/container-license-acceptance.txt [hunk 2]
```diff
@@ -0,0 +1 @@
+taskana/db2:11.5
```
Comment (mustaphazorgati) [inline lib/taskana-core-test/src/test/resources/container-license-acceptance.txt:1]: this should be part of the test-api. Can you move this there?

File: lib/taskana-core-test/src/test/java/acceptance/task/query/TaskQueryImplAccTest.java (hunk 1 above)
Comment (mustaphazorgati) [inline lib/taskana-core-test/src/test/java/acceptance/task/query/TaskQueryImplAccTest.java:263]: Did you manually add this, or did you IDE format this?

File: lib/taskana-core-test/src/test/resources/container-license-acceptance.txt (hunk 2 above)
Comment (mustaphazorgati) [inline lib/taskana-core-test/src/test/resources/container-license-acceptance.txt:1]: this should be part of the test-api. Can you please move it there?

File: lib/taskana-test-api/src/main/java/pro/taskana/testapi/DefaultTestEntities.java [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+import java.util.UUID;
+
+import pro.taskana.classification.internal.builder.ClassificationBuilder;
+import pro.taskana.task.internal.builder.ObjectReferenceBuilder;
+import pro.taskana.workbasket.api.WorkbasketType;
+import pro.taskana.workbasket.internal.builder.WorkbasketBuilder;
+
+public class DefaultTestEntities {
+
```
Comment (mustaphazorgati) [inline lib/taskana-test-api/src/main/java/pro/taskana/testapi/DefaultTestEntities.java:11]: please add a private default constructor like this:
```
private DefaultTestEntities(){
    throw new IllegalStateException("Utility class");
}

File: lib/taskana-core-test/src/test/java/acceptance/task/update/UpdateManualPriorityAccTest.java [hunk 4]
```diff
@@ -0,0 +1,176 @@
+package acceptance.task.update;
```
[the remark this answers is not in the document]
Comment (mustaphazorgati) [reply lib/taskana-core-test/src/test/java/acceptance/task/update/UpdateManualPriorityAccTest.java:1]: This seems to be a duplicate. Please remove.

File: lib/taskana-core-test/src/test/java/acceptance/task/update/UpdateManualPriorityWithSpiAccTest.java [hunk 5]
```diff
@@ -0,0 +1,168 @@
+package acceptance.task.update;
```
[the remark this answers is not in the document]
Comment (mustaphazorgati) [reply lib/taskana-core-test/src/test/java/acceptance/task/update/UpdateManualPriorityWithSpiAccTest.java:1]: This seems to be a duplicate. Please remove.

**Review by mustaphazorgati — CHANGES_REQUESTED** [review]
General comment:
Please remove `lib/taskana-core-test/src/main/java/.gitignore`
`taskana-core-test` is not supposed to have a `src` directory.

File: lib/taskana-core-test/src/test/java/acceptance/task/query/TaskQueryImplAccTest.java (hunk 1 above)
Comment (ryzheboka) [reply lib/taskana-core-test/src/test/java/acceptance/task/query/TaskQueryImplAccTest.java:263]: The IDE formatted it like this. Should I format it back manually?

Comment (ryzheboka) [reply lib/taskana-core-test/src/test/java/acceptance/task/query/TaskQueryImplAccTest.java:263]: This seems to be the same line in the same file as above.

File: lib/taskana-core/src/test/java/acceptance/task/update/UpdateTaskWithSorAccTest.java [hunk 6]
```diff
@@ -1,169 +0,0 @@
-package acceptance.task.update;
```
[the remark this answers is not in the document]
Comment (ryzheboka) [reply lib/taskana-core/src/test/java/acceptance/task/update/UpdateTaskWithSorAccTest.java:1]: It was an accident, I will add it to taskana-core-test

File: lib/taskana-core-test/pom.xml [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+      <groupId>pro.taskana</groupId>
+      <artifactId>taskana-common-logging</artifactId>
+      <version>${project.version}</version>
+    </dependency>
+    <dependency>
+      <groupId>pro.taskana</groupId>
+      <artifactId>taskana-core</artifactId>
+      <version>${project.version}</version>
+    </dependency>
+    <dependency>
```
[the remark this answers is not in the document]
Comment (ryzheboka) [reply lib/taskana-core-test/pom.xml:28]: Ok :)

File: lib/taskana-core-test/src/test/resources/container-license-acceptance.txt (hunk 2 above)
[the remark this answers is not in the document]
Comment (ryzheboka) [reply lib/taskana-core-test/src/test/resources/container-license-acceptance.txt:1]: Ok :)

File: lib/taskana-core-test/src/test/java/acceptance/task/query/TaskQueryImplAccTest.java (hunk 1 above)
Comment (ryzheboka) [reply lib/taskana-core-test/src/test/java/acceptance/task/query/TaskQueryImplAccTest.java:263]: Ok :)

**Review by mustaphazorgati — CHANGES_REQUESTED** [review]
I was able to replace the `.gitkeep` files with sonar property adjustments.
I've sent you the patch file through slack. Please apply those changes :)

**Review by mustaphazorgati — APPROVED** [review]

**Pull request merged**